pkg/core/pipeline/condition/main.go

```diff
@@ -84,7 +85,25 @@ func (c *Condition) Run(source string) (err error) {
 			return err
 		}
 	} else {
-		return fmt.Errorf("something went wrong while looking at the scm configuration: %v", c.Config.Scm)
+		var scmKind string
+		for key := range c.Config.Scm {
+			scmKind = key
+		}
+		switch scmKind {
+		case "github":
+			interf := reflect.ValueOf(c.Config.Scm["github"]).Interface()
+			redacted := interf.(map[string]interface{})
+			redacted["token"] = "********"
+			return fmt.Errorf("something went wrong while looking at the scm configuration: %+v", redacted)
+		case "git":
+			interf := reflect.ValueOf(c.Config.Scm["git"]).Interface()
+			redacted := interf.(map[string]interface{})
+			redacted["password"] = "********"
+			return fmt.Errorf("something went wrong while looking at the scm configuration: %+v", redacted)
+		default:
```

Comment:
This PR is really a good idea and really useful!

However, I have a bad feeling about maintaining a additional list of possible SCM implementations here.
Also bad feeling about using reflect 😅

WDYT about defining a method `ToString()` for the Scm interface? Then you call this method in the log method

```
fmt.Errorf("something went wrong while looking at the scm configuration: %s", c.Config.Scm.ToString())
```


And then you delegate the implementation of the method to each SCM plugin: no need for reflect, no need for a if here

Reply:
I'll need help, I don't see how to do it.

Reply:
```
fmt.Errorf("something went wrong while looking at the scm configuration: %s", c.Config.Scm.ToString())
```
I didn't managed to access `.ToString()`, as here `c.Config.Scm` seems to be a map instead of a ScmHandler (?)

Reply:
It is a map of interface (with keys of type string) as per https://github.com/updatecli/updatecli/blob/952fbdd4d2df163b13d852c06562b74f10e31ee2/pkg/core/pipeline/resource/main.go#L36. It means that you might have to cast the generic `interface{}` type into a `scm.ScmHandler{}` interface type.


I wonder if this "else" case is still valid though: it means that there is no valid SCM associated to the current condition. Not sure why would we check the deprecated stuff (should be the job of the `deprecated.go` file).

Reply:
The `Git` type has all needed properties, while the `Github` one has an intermediate `Spec` I didn't managed to access the same way.

Reply:
I don't understand what you are saying: since the `ScmHandler{}` interface defines a `ToString()` method, then any SCM implementation that fulfill this interface must implement this method. 
So if you "cast" the value of the map to an `ScmHandler{}` type, then you should be able to call the method.

Reply:
How can I cast it to an `ScmHandler`?

> I wonder if this "else" case is still valid though: it means that there is no valid SCM associated to the current condition. Not sure why would we check the deprecated stuff (should be the job of the deprecated.go file)

I've got bitten by it, I'll have to remember how though ^^